### DISCUSSION TITLE
Better Class Identity through practical challenges, missions and game mechanics

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#4971] [Ideas and suggestions]
Barotrauma has defined 4 submarine classes for its vanilla sub: Scout, Attack, Transport, and Deep Diver.
Those classes can be selected in the editor for the user-made submarines. While these might just be flavor, when looking deeper into it...

1. **There is a "class identity crisis" going on right now.** 

To explain that, I'm going to use two complementary concepts of identity, one that comes from the top-down (Thematic) and one that arises from the bottom-up (Practical)

**Thematic identity** is the range of ideas and notions the name of the class evokes to players

**Practical identity** is how the game mechanics push sub builders to specialize their vessels, as an answer to a given challenge

Both go hand in hand to define class identity, but unfortunately there is an imbalance: Thematic is cheap, but Practical relies on more complex elements and game mechanics, as well as game knowledge. As a result of this imbalance, the game doesn't have enough Practical identity to match its Thematic counterpart, and in one particular case, isn't even sure of its own Thematic.

What is the Thematic of the 4 classes? Here's what comes to mind:

- **Scout**: small, agile, fast, reconnaissance... made for moving around, fast, efficiently, and exploring/spotting things. 
- **Transport**: large, bulky, can carry cargo and/or people from point A to point B safely, making one or several trips
- **Attack**: powerful, offensive, weapons, guns, ammunition... it kills and destroys things
- **Deep Diver**: Abyss, exploration, darkness, pressure? _We might have a problem here, the concept is somewhat vague, and it will show later_

This is my visualization of what all those 4 classes mean, but rather than my interpretation, how it's interpeted in the vanilla submarines _should_ give a clearer indication of what it all means in **Practical** terms. But does it? Unfortunately not. Scouts while decently fast, lack basic equipment, and predate cave and wreck updates, making them ill equipped for exploration. Transport submarine have huge, empty cargo bays that serve no purpose when Transport missions ask at most two crate slots. Deep Diver (singular) Remora is a confused submarine, with no clear purpose. Only attack crafts actually fulfill their role, as the game constantly throws enemies at the player (regardless of class). **The game doesn't have good Practical identity for its submarines**

2. Instead, it's trying to push a rather forceful solution to push class identity: **Class Exclusivity**. 

The first iteration of "exclusivity" was the introduction of a passive, 20% crushing depth resistance, exclusive to "Deep Diver" subs; other classes will probably get their own as well. There are also plans to introduce Class-exclusive missions, such as bigger loads for transport subs (to my understanding), a scaling extension of the current transport mission.

This approach however, does not address the lack of "Practical". Submarines might get better at some basic function, but there is no strong differentiation between roles. The game doesn't pose enough "problems", so players aren't incentivized to come up with specialized solutions, and as a result all submarines become functionally same-y.

You might make the same submarine model, but in 4 different class categories, and switch to the alternate version the bonus that suits you most depending on the situation. A gigantic submarine can be classified as Scout, just because the Scout "class bonus" would be more appealing than the Transport bonus. This kills the Thematic meaning of the classes; you're just picking a perk out of 4.

3. **Attempting a Practical approach to class identity**

Let's try pursuing the opposite approach. Rather than applying exclusive benefits to submarines through hard-coded, innate benefits, let's leave the approach to the sub builders, and instead nudge builders towards intuitive designs that are _solutions_ to practical problems. Thus the title: _Class Identity through practical challenges, missions and game mechanics_

How, and what does that mean? Use pre-existing game mechanics, and come up with "challenges", of which previously _Thematic_ identity is a direct answer. This could mean enemies, the environment, or specific objectives. The goal isn't to create exclusive content, as those challenges will be available to every class, but rather reward specialized submarine building in an organic way.

**Important note: All of the following suggestions aren't class restricted.** Rather, every submarine class can tackle those challenges, but submarines following the class thematic will perform better in practical gameplay aspects. 

**Scout**: The practical advantages of a Scout submarine would be its small size, its speed, as well as maneuvrability. Therefore we need challenges where the nimbler, smaller submarines excel:

![Baroclassscout](https://user-images.githubusercontent.com/73229309/106815764-9b240280-6674-11eb-893f-6c6ab682c9a5.png)

1. AGILITY: Paths with large amounts of floating icebergs, or generating caves with obstructed passages and dead ends. Scouts are naturally better at moving around obstacles and back in the path. Other, bulkier submarines would waste energy and ammunition breaking the obstacles instead
2. SPEED: Limited time missions with an exploration component. Going fast, accelerating and navigating close to cave walls, but also the ability to split the crew using small exploration shuttles to increase the chances of finding the probe in time. Bigger submarines are slower and risk hitting the surrounding terrain.
3. NAVIGATION: Finding the shortest path and using precise movement to optimize the number of relays. The mission demands good control and exploration capabilities, and even possibly using the smaller side paths that larger subs cannot take.

**Transport**: Transport submarines are large and can hold large amounts of items, but lack mission structures that _truly_ demand spacious rooms and specialized design. Both the Berilia and the R-29 are too big for no real purpose, considering transport missions only consist of carrying two crates in an uneventful manner. As such, the idea of transportion in Baro needs a rework:

![Baroclasstransport](https://user-images.githubusercontent.com/73229309/106816327-8005c280-6675-11eb-8743-fbe9e9f9ab17.png)

1. CARGO: Replacing the very nature of transport missions by introducing standardized, physical cargo containers that attach to your submarine at designated ports, and that must be kept safe from external dangers. A true Transport-class submarine would have the builder designate and build protected cargo bays as part of the whole, moving the sub and its cargo as one. Other submarines would carry the container externally, exposing it and impeding mobility and defensive capabilities.
2. TRANSIT: Transporting passengers would require specific room design to secure their well-being via a "Stress" system. Crew/Recreation quarters are often a luxury, taking a good ammount of non-functional space in most submarines. Transport subs, due to their large size, could afford to carry NPCs comfortably to their destination. Other, smaller or more functional submarines would stress out the passengers and lower the payout.
3. DELIVERY: Introducing a form of "opportunity" trade, where players would be able to load up large amounts of crates for a long duration, delivering them across several places. This takes up a considerable amount of crate slots; Transport submarines can stock up on crates, while smaller submarines would avoid sacrificing valuable storage space.

**Attack:** If there's one thing the game does well, it's throwing enemies at the player. As such, we've seen a good variety of weapons being created, and powerful submarines. Attack submarines are, as such, inherently useful. All sorts of armaments have a purpose, as there is no shortage of enemies. _Therefore, nothing to see here!_

**Deep Diver: We have a problem**: The tragedy of Deep Diver, or DD, is that it has a somewhat elusive, poorly theme with only one in-game submarine, the Remora, belonging to the class. _What is a DD?_ is the question that needs to be answered; does the game push the players to ALWAYS go deeper? If going deeper is systemic to progression, then every submarine has to eventually become a deep diver, but if all submarines are DDs, it makes the very concept of DD meaningless. Upgrading the hull to fight back "crushing depth" doesn't actually add a "depth" mechanic, it eliminates it the moment you purchase said upgrades. It is merely an economic gate to slow down progress. As a result, I'll have to substitute this system with a new one:

![BaroclassDD](https://user-images.githubusercontent.com/73229309/106818725-f952e480-6678-11eb-998d-5a35dc5d36ef.png)

1. DEEP BIOME: There is a permanently accessible "depths" biome at the bottom of the map. It is a shorter, more navigable path than the shallower, regular caves, but it comes with two dangerous twists that heavily incentivize specialization: 
2. ABYSSALS: Enemies from the deep are more dangerous, their shells are thicker and their claws are sharper. However, they don't see as good, and will react to noise and lights. Navigating past them is possible if the submarine is kept "silent"; by turning off machinery, power, lights, sonar, reducing engine speed and using proximity chat over radio. Submarines incoporating this kind of silent running would stealth their way past abyssals, while others might get into a fight.
3. PRESSURE: Crushing depth would constantly affect the submarines in the depths, creating unpredictable leaks. A properly made Deep Diver would have better flood management than most submarines; better compartments, better draining, better control of ducts that keep them afloat more easily.

Unlike Scout and Transport, these Deep Diver suggestions are hardly final, considering how loosely defined the concept is, so it's still very open to imagination, but the class needs a lot of work on both fronts to come into its own.

---------------------------------------------------------------------------

In conclusion, class identity is something that should arise from the ingenuity of sub builders, when faced with practical challenges throughout the campaign. It should not be akin to a "class bonus", but instead a physical consequence of specific design choices: is the submarine small? is it big? is it well armed? is it fast? is it specialized? Those practical challenges must provide advantages and disadvantages to different builds. 

With such an incentive, classes would finally gain proper meaning; even if they are simply flavor labels

**Comment by mLuby** [comment]
To be pedantic: this refers to [ship types](https://en.wikipedia.org/wiki/List_of_ship_types) (sedan, minivan, SUV, pickup truck), not [ship classes](https://en.wikipedia.org/wiki/Ship_class) (Prius, Odyssey, Escalade, F-150).

> A ship class is a group of ships of a similar design. This is distinct from a ship type, which might reflect a similarity of tonnage or intended use. For example, USS Carl Vinson is a nuclear aircraft carrier (ship type) of the Nimitz class (ship class). -[Wikipedia](https://en.wikipedia.org/wiki/Ship_class)

**Comment by Hinny** [comment]
I like this a lot! Good job! I hope that the devs take inspiration from this.
I just want to add that besides better flood plan on deep divers, I hope they explore a dynamic way of calculating how much pressure a sub / compartment can withstand. Maybe by only having small rooms without windows and rooms with support beams a sub can further specialize in deep diving.

**Comment by fracter2** [comment]
I like this idea a lot
its well thought out and formulated, and it would fit the game very nicely as far as i can tell
